plugins/janus_audiobridge.c

```diff
@@ -2605,6 +2610,49 @@ json_t *janus_audiobridge_query_session(janus_plugin_session *handle) {
 	return info;
 }
 
+static int janus_audiobridge_access_room(json_t *root, gboolean check_modify, janus_audiobridge_room **audiobridge, char *error_cause, int error_cause_size) {
+	/* rooms_mutex has to be locked */
+	int error_code = 0;
+	json_t *room = json_object_get(root, "room");
+	guint64 room_id = 0;
+	char room_id_num[30], *room_id_str = NULL;
+	if(!string_ids) {
+		room_id = json_integer_value(room);
+		g_snprintf(room_id_num, sizeof(room_id_num), "%"SCNu64, room_id);
+		room_id_str = room_id_num;
+	} else {
+		room_id_str = (char *)json_string_value(room);
+	}
+	*audiobridge = g_hash_table_lookup(rooms,
+		string_ids ? (gpointer)room_id_str : (gpointer)&room_id);
+	if(*audiobridge == NULL) {
+		JANUS_LOG(LOG_ERR, "No such room (%s)\n", room_id_str);
+		error_code = JANUS_AUDIOBRIDGE_ERROR_NO_SUCH_ROOM;
+		if(error_cause)
+			g_snprintf(error_cause, error_cause_size, "No such room (%s)", room_id_str);
+		return error_code;
+	}
+	if(g_atomic_int_get(&((*audiobridge)->destroyed))) {
+		JANUS_LOG(LOG_ERR, "No such room (%s)\n", room_id_str);
+		error_code = JANUS_AUDIOBRIDGE_ERROR_NO_SUCH_ROOM;
+		if(error_cause)
+			g_snprintf(error_cause, error_cause_size, "No such room (%s)", room_id_str);
+		return error_code;
+	}
+	// FIXME:: it should check for recording lock
```

Comment:
Code style: comments must always be in `/* */` form, not `//`.
Not sure why the comment says that, though: if this method was copied from the VideoRoom, then it can be used for other things as well, not just the new recording API you added.

Reply:
done